### PR TITLE
feat: [CODE-1227]: plg support for CODE

### PR DESCRIPTION
**Pull request by rkapoor10**
### Summary

Added PLG support for CODE module
for more details - [CODE-1227](https://harness.atlassian.net/browse/CODE-1227) and [ CODE PLG Support](https://harness.atlassian.net/wiki/spaces/CODE/pages/21584052277/PLG+Support)

- license support for CODE Repository
- added card for new user license on signup
- CODEHomePage
- CODETrialHomePage
- ModuleSelect Card

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[CODE-1227]: https://harness.atlassian.net/browse/CODE-1227?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/modules/60-code/RouteDefinitions.ts [hunk 1]
```diff
@@ -43,6 +43,23 @@ const REPOS_PREFIX = '/repos'
 const CODE = 'code'
 
 export const routesV2 = {
+  // toCODE: routes.toCODE,
```
Comment (tan-nhu) [inline src/modules/60-code/RouteDefinitions.ts:46]: Why this entry is commented out?

Comment (rkapoor10) [reply src/modules/60-code/RouteDefinitions.ts:46]: not added to V2, as it was not required. I'll remove it.

**Comment by rkapoor10** [comment]
retrigger coverage

**Comment by rkapoor10** [comment]
retrigger jest

**Comment by rkapoor10** [comment]
retrigger jest

**Review by tan-nhu — APPROVED** [review]

**Review by hitesharinga — APPROVED** [review]

**Pull request merged**